GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -313,7 +313,20 @@ public FormBrowse([NotNull] GitUICommands commands, string filter, ObjectId sele
             UICommands.BrowseRepo = this;
             _controller = new FormBrowseController(new GitGpgController(() => Module));
             _commitDataManager = new CommitDataManager(() => Module);
+
+            toolStripButtonLevelUp.Visible = false;
             _submoduleStatusProvider = new SubmoduleStatusProvider();
+            _submoduleStatusProvider.SubmodulesChanged += (sender, e) =>
+            {
+                ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                {
+                    await this.SwitchToMainThreadAsync();
+
+                    toolStripButtonLevelUp.Visible = e.HasSubmodules || e.HasSupermodule;
+                    toolStripButtonLevelUp.Image = e.HasSupermodule ? Images.NavigateUp : Images.SubmodulesManage;
```

Comment:
Images.SubmodulesManage I prefer Images.FolderSubmodule in this context, hard to see Sucmodule manage here.
Images.NavigateUp Is this better? Hesitant
None Hesitant

What to use in the documentation?

Reply:
On the icon choice, I find the icon with the blue background too visually dominant relative to the other toolbar icons. Is it really too hard to see now, or would it just be a matter of getting used to the change? There may be a middle ground. Interested to hear other people's opinions too. 

Regarding the icon change, my concern is that the button sometimes navigates upwards, sometimes opens the submodule. It's not clear from the icon when to expect either behaviour. I would also be happy to have 'navigate to supermodule' become a submenu item.

> What to use in the documentation?

I don't follow, sorry.

Reply:
> Is it really too hard to see now, or would it just be a matter of getting used to the change? 

I know where to click anyway and can get used to it. Just voting against the change.

> I don't follow, sorry.

When we update the documentation, contextual need to be described somehow.
To answer this myself: Use "submodule" icon in the screenshot, describe the uparrow and removal.

It is better to merge PRs like this before anyone start to update the documentation.

Reply:
Thanks for clarifying around the docs. Sounds like we need a cutoff for changes like this, so we can get the translation and documentation phases underway without wasting effort. If this PR doesn't make the bar, I can retarget it to a 3.1 branch or similar (or maybe we make a 3.0 branch and keep master for 3.1).

Reply:
I would avoid multiple branches, if possible. Reconciliation can be nightmarish.
v3 is not far away, after that we should be back to normal.